dashboard/app/models/pd/workshop.rb

```diff
@@ -185,19 +185,36 @@ def self.send_reminder_for_upcoming_in_days(days)
     end
   end
 
-  def self.send_exit_surveys
-    # TODO: handle teachers who show up, join the section, and are marked attended without enrolling.
-    end_in_days(0).each do |workshop|
-      workshop.enrollments.each do |enrollment|
-        teacher = User.find_by(email: enrollment.email, user_type: User::TYPE_TEACHER)
-        Pd::WorkshopMailer.exit_survey(workshop, teacher, enrollment).deliver_now
-      end
-    end
-  end
-
   def self.send_automated_emails
     send_reminder_for_upcoming_in_days(3)
     send_reminder_for_upcoming_in_days(10)
-    send_exit_surveys
+  end
+
+  def self.process_closed_workshop_async(id)
+    workshop = Pd::Workshop.find(id)
```

Comment:
Actually there is no `find!`. See http://api.rubyonrails.org/classes/ActiveRecord/FinderMethods.html#method-i-find

"ActiveRecord::RecordNotFound will be raised if one or more ids are not found." (no comment on ActiveRecord's inconsistent method naming). I could use `find_by_id!` to be more explicit?